File: Server/Packet.py
# 패킷 

# Client -> Server 

Login = "LOGIN"
Loginlist = "LOGINLIST"
Logout = "LOGOUT"
Shortmessage = "SHORTMESSAGE"
Sendfile = "SENDFILE"
Sendbyte = "SENDBYTE"

# Server -> Client

Login_ACK = "LOGIN_ACK"
Loginlist_ACK = "LOGINLIST_ACK"
Logout_ACK = "LOGOUT_ACK"
Shortmessage_ACK = "SHORTMESSAGE_ACK"
Sendfile_ACK = "SENDFILE_ACK"
Sendbyte_ACK = "SENDBYTE_ACK"

def LogIn_ACK(name):
    pack = ''

    pack += Login_ACK + '@'
    pack += name

    return pack

def LogInList_ACK(user_list):
    pack = ''

    pack += Loginlist_ACK + '@'
    for user in user_list:
        pack += user + '#'

    return pack

def LogOut_ACK(name):
    pack = ''

    pack += Logout_ACK + '@'
    pack += name

    return pack

def ShortMessage_ACK(name, msg):
    pack = ''

    pack += Shortmessage_ACK + '@'
    pack += name + '#'
    pack += msg

    return pack

def FileSend_ACK(filename, size):
    pack = ''

    pack += Sendfile_ACK + '@'
    pack += filename + '#'
    pack += size

    return pack

def BytesSend_ACK(bytes):
    pack = ''

    decoded_bytes = bytes.decode('eur-kr')
    
    pack += Sendbyte_ACK + '@'
    pack += decoded_bytes

    return pack

File: Project_Server/Project_Server/Server/Packet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Server
{
    internal static class Packet
    {
        // Client -> Server
        public const string Login = "LOGIN";
        public const string Logout = "LOGOUT";
        public const string Shortmessage = "SHORTMESSAGE";
        public const string Sendfile = "SENDFILE";
        public const string Sendbyte = "SENDBYTE";
        public const string Sendremote = "SENDREMOTE";

        // Server -> Client
        public const string Login_ACK = "LOGIN_ACK";
        public const string Logout_ACK = "LOGOUT_ACK"; 
        public const string Shortmessage_ACK = "SHORTMESSAGE_ACK";
        public const string Sendfile_ACK = "SENDFILE_ACK";
        public const string Sendbyte_ACK = "SENDBYTE_ACK";
        public const string Sendremote_ACK = "SENDREMOTE_ACK";

        //메시지(채팅)
        public static string LogIn_ACK(string name)
        {
            string packet = string.Empty;

            packet += Login_ACK + '@';

            packet += name;

            return packet;
        }

        public static string LogOut_ACK(string name)
        {
            string packet = string.Empty;

            packet += Logout_ACK + '@';

            packet += name;

            return packet;
        }


        public static string ShortMessage_ACK(string nickname, string msg)
        {
            string packet = string.Empty;

            packet += Shortmessage_ACK + '@';

            packet += nickname + '#';
            packet += msg;

            return packet;
        }

        //파일 보내기
        public static string SendFile_ACK(string filename, int size)
        {
            string packet = string.Empty;

            packet += Sendfile_ACK + '@';

            packet += filename + '#';
            packet += size;

            return packet;
        }

        //바이트 배열 보내기( 화면공유 )
        public static string SendByte_ACK(byte[] bytes)
        {
            string packet = string.Empty;

            packet += Sendbyte_ACK + '@';

            packet += bytes;

            return packet;
        }

        //원격 제어
        public static string SendRemote_ACK(byte[] bytes)
        {
            string packet = string.Empty;

            packet += Sendremote_ACK + '@';

            packet += bytes;

            return packet;
        }
    }
}